File: Kathmandu University/C++/hello_world.cpp
#include <iostream>
#include <string>

int main() {
    std::cout << "Enter your name: ";
    std::string name;
    std::getline(std::cin, name);

    if (name.empty()) {
        std::cout << "Hello, mysterious stranger!" << std::endl;
    } else {
        std::cout << "Hello, " << name << "!" << std::endl;
    }

    return 0;
}

File: Kathmandu University/Java/helloworld.java
import java.util.Scanner;

public class HelloWorld {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter your name: ");
        String name = scanner.nextLine().trim();
        if (name.isEmpty()) {
            System.out.println("Hello, World from Java!");
        } else {
            System.out.printf("Hello, %s! Welcome to Java.%n", name);
        }
        scanner.close();
    }
}
